bash-preexec.sh

```diff
@@ -39,8 +39,12 @@ fi
 __bp_imported="defined"
 
 # Should be available to each precmd and preexec
-# functions, should they want it.
+# functions, should they want it. $? and $_ are available as $? and $_, but
+# $PIPESTATUS is available only in a copy, $BP_PIPESTATUS.
+# TODO: Figure out how to restore PIPESTATUS before each precmd or preexec
+# function.
 __bp_last_ret_value="$?"
+BP_PIPESTATUS=("${PIPESTATUS[@]}")
```

Comment:
I'd suggest `__bp_pipestatus` instead of `BP_PIPESTATUS` to be consistent with other "private" variables bash-preexec uses.

Reply:
I chose `BP_PIPESTATUS` to make this explicitly public, since I couldn't think of another way to pass it to pre{exec,cmd} functions.

Reply:
Ah sorry, I didn't realize this was intentional. If it's part of the public API it should probably be documented (e.g. in the README).